### ISSUE TITLE
Galaxy collection import is hanging in the "waiting" status forever

### ISSUE THREAD

**Issue by spetrosi** [ansible/galaxy#2740]
##### SUMMARY
The import for the microsoft.sql collection is hanging in the waiting state for more than 2 hours without any state updates. Usually the import happens instantly. 

##### STEPS TO REPRODUCE
In the web UI, I click *My Content*, expand *microsoft*, click *Upload New Version*, select the attached tarball `microsoft-sql-1.0.12.tar.gz`. 

##### EXPECTED RESULTS
Galaxy successfully imports the new collection version.

##### ACTUAL RESULTS
The import hangs in the waiting state.

[microsoft-sql-1.0.12.tar.gz](https://github.com/ansible/galaxy/files/6791690/microsoft-sql-1.0.12.tar.gz)

**Comment by Jason-Zhang9309** [comment]
I am in the same situation as you.
Still waiting now?
Do you have any ideas to solve the problem？

**Comment by spetrosi** [comment]
No idea, my import is still in the "waiting" state since 3 days ago.

**Comment by Jason-Zhang9309** [comment]
Thanks for your response.
It's kind of embarrassing that there is no way to cancel it or timeout limit.

**Comment by cityofships** [comment]
Got the same

**Comment by rossnet** [comment]
We have the same issue since Friday afternoon CET. An official update on the situation would be very much appreciated!

**Comment by dominikkaminski** [comment]
Good news: Now it seems to be working normally again. I've already restarted all my failed import without any issues :partying_face:

**Comment by spetrosi** [comment]
I have found out that my collection tarball was missing a meta/runtime.yml file. I saw an error message about this when trying to import the artifact again. After I added runtime.yml import worked fine.

**Comment by cityofships** [comment]
> I have found out that my collection tarball was missing a meta/runtime.yml file. I saw an error message about this when trying to import the artifact again. After I added runtime.yml import worked fine.

Noticed that too. Is there an official statement that this is now required? As per documentations, so far this was optional: https://docs.ansible.com/ansible/devel/dev_guide/developing_collections_structure.html#meta-directory

**Comment by jobcespedes** [comment]
> I have found out that my collection tarball was missing a meta/runtime.yml file. I saw an error message about this when trying to import the artifact again. After I added runtime.yml import worked fine.

I think it was a secondary issue aside from what happened here, because we have `meta/runtime.yml` in place and still the build got stuck over the weekend.

**Comment by thedoubl3j** [comment]
@spetrosi @jobcespedes @dominikkaminski the worker nodes were hanging and we had to kill then restart them. this also caused any in flight uploads that were in a waiting status to be cancelled. Uploads (though I am late to responding to this) should be working correctly. 

The meta/runtime.yml file and its subsequent need was added to the importer for various reasons, primarily for documentation purposes and compat reasons. the lack of announcement I can't directly speak to aside from a lot of moving parts/people. our apologies. any changes going forward, I will work with @gundalow and co to see if we can get them published in the Bullhorn (https://github.com/ansible/community/issues/546)
